src/main/java/io/kokuwa/keycloak/metrics/store/MicrometerEventStoreProviderFactory.java

```diff
@@ -0,0 +1,51 @@
+package io.kokuwa.keycloak.metrics.store;
+
+import io.kokuwa.keycloak.metrics.CommunityProfiles;
+import io.kokuwa.keycloak.metrics.event.MetricsEventListenerFactory;
+import org.jboss.logging.Logger;
+import org.keycloak.Config;
+import org.keycloak.connections.jpa.JpaConnectionProvider;
+import org.keycloak.events.EventStoreProvider;
+import org.keycloak.events.jpa.JpaEventStoreProviderFactory;
+import org.keycloak.models.KeycloakSession;
+
+import java.util.stream.Collectors;
+
+public class MicrometerEventStoreProviderFactory extends JpaEventStoreProviderFactory {
+    private static final Logger log = Logger.getLogger(MicrometerEventStoreProviderFactory.class);
+
+    public static final String ID = "jpa"; // Override default event store provider
+    private boolean replaceIds;
+
+    @Override
+    public EventStoreProvider create(KeycloakSession session) {
+        //removeMetricsEventsListenerIfExists(session);
+
+        JpaConnectionProvider connection = session.getProvider(JpaConnectionProvider.class);
+        return new MicrometerEventStoreProvider(session, connection.getEntityManager(), replaceIds);
+    }
+
+    public void init(Config.Scope config) {
+        replaceIds = "true".equals(System.getenv().getOrDefault("KC_METRICS_EVENT_REPLACE_IDS", "true"));
+        log.info(replaceIds ? "Configured with model names." : "Configured with model ids.");
+    }
+
+    @Override
+    public String getId() {
+        return ID;
+    }
+
+    // we need to remove 'metrics-listener' if already exists in realm, and KC_COMMUNITY_EVENTS_METRICS_ENABLED=true otherwise the keycloak_1  | 2024-06-22 06:37:29,059 ERROR [org.keycloak.services] (executor-thread-3) KC-SERVICES0083: Event listener 'metrics-listener' registered, but provider not found
+    //Todo: This approach gives Unique index or primary key violation: "PUBLIC.PRIMARY_KEY_C38 ON PUBLIC.REALM_EVENTS_LISTENERS(REALM_ID, ""VALUE"") VALUES ( /* key:4 */ '32497f09-6079-42d3-8f56-9e4654b53e5e', 'jboss-logging')"; SQL statement:
+    //Todo: Approach is same on AdminRealmEvents.updateRealmEventsConfig. xgp any idea why?
+    private static void removeMetricsEventsListenerIfExists(KeycloakSession session) {
```

Comment:
Rather than "removing" it, since we control the listener, can't we set some kind of variable in the session (e.g. a session note), check that in the listener, don't run it if it's set, and then unset the variable?

Reply:
The case I was thinking:
1. In the initial state of the KC instance KC_COMMUNITY_EVENTS_METRICS_ENABLED=false
2. A admin adds the 'metrics-listener'  in the Events listener tab
3. The KC instance KC_COMMUNITY_EVENTS_METRICS_ENABLED flag is set to true
4. The admin goes to the Events tab an gets ' Event listener 'metrics-listener' registered, but provider not found' 

This is because the MetricsEventListenerFactory implements EnvironmentDependentProviderFactory https://github.com/p2-inc/keycloak-event-metrics/pull/1/files#r1649953987. 

I think at this point we no longer control this MetricsEventListener right?